filebeat/prospector/prospector_log.go

```diff
@@ -171,8 +173,18 @@ func (p *ProspectorLog) harvestExistingFile(newState file.State, oldState file.S
 		// One problem with comparing modTime is that it is in seconds, and scans can happen more then once a second
 		logp.Debug("prospector", "Resuming harvesting of file: %s, offset: %v", newState.Source, oldState.Offset)
 		p.Prospector.startHarvester(newState, oldState.Offset)
+		return
+	}
+
+	// File size was reduced -> truncated file
+	if oldState.Finished && newState.Fileinfo.Size() < oldState.Offset {
+		logp.Debug("prospector", "Old file was truncated. Starting from the beginning: %s", newState.Source)
```

Comment:
I added a counter. A note here: either the harvester can detect it if it is an open file or the prospector if is is a non actively harvested file. Should we use 2 different counters?